src/containers/Tasks/RequestTerms/RequestTerms.tsx

```diff
@@ -47,35 +43,36 @@ class RequestLegitimation extends React.Component<
     }
   }
 
-  private isValid(): number {
-    const { receiverAddresses } = this.props
-    return receiverAddresses && receiverAddresses.length
-  }
-
   private handleSubmit(): void {
     const { cTypeHash, receiverAddresses, onFinished } = this.props
     const { selectedClaimEntries } = this.state
-    let claims: sdk.IPartialClaim[] = cTypeHash ? [{ cTypeHash }] : []
+    let claims: sdk.IPartialClaim[] = [{ cTypeHash } as sdk.IPartialClaim]
 
     if (selectedClaimEntries && selectedClaimEntries.length) {
-      claims = selectedClaimEntries.map(claimEntry => claimEntry.claim)
+      claims = selectedClaimEntries.map(
+        (claimEntry: Claims.Entry) => claimEntry.claim
+      )
     }
 
     if (this.isValid()) {
-      attestationWorkflow
-        .requestLegitimations(claims, receiverAddresses)
-        .then(() => {
-          if (onFinished) {
-            onFinished()
-          }
-        })
+      attestationWorkflow.requestTerms(claims, receiverAddresses).then(() => {
+        if (onFinished) {
+          onFinished()
+        }
+      })
     }
   }
 
+  private isValid(): any {
+    const { receiverAddresses } = this.props
+    return receiverAddresses && receiverAddresses.length
+  }
```

Comment:
this gets the type right:
```suggestion
  private isValid(): boolean {
    const { receiverAddresses } = this.props
    return !!(receiverAddresses && receiverAddresses.length)
  }
```

Reply:
Thank you, updated.